src/main/java/com/ntnghia/task/service/impl/TaskServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.ntnghia.task.service.impl;
+
+import com.ntnghia.task.entity.Task;
+import com.ntnghia.task.exception.NotFoundException;
+import com.ntnghia.task.repository.TaskRepository;
+import com.ntnghia.task.service.TaskService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class TaskServiceImpl implements TaskService {
+    @Autowired
+    private TaskRepository taskRepository;
+
+    @Override
+    public List<Task> getAll() {
+        return taskRepository.findAll();
+    }
+
+    @Override
+    public Task findById(int id) {
+        if (taskRepository.findById(id).isPresent()) return taskRepository.findById(id).get();
+
+        throw new NotFoundException(String.format("Task id %d not found", id));
+    }
+
+    @Override
+    public List<Task> findByKeyword(String keyword) {
+        return taskRepository.findByKeyword(keyword);
+    }
+
+    @Override
+    public Task saveTask(Task task) {
+        return taskRepository.save(task);
+    }
+
+    @Override
+    public Task updateTask(int id, Task task) {
+        if (taskRepository.existsById(id)){
+            task.setId(id);
+            return taskRepository.save(task);
+        }
+
+        throw new NotFoundException(String.format("Task id %d not found", id));
```

Comment:
Thử bỏ đi cái `String.format(` bằng cách thay đổi param của class NotFoundException

Reply:
Em nghĩ là nên vậy cho dễ tái sử dụng exception này. Ví dụ not found name thì mình print ra name nữa.